### ISSUE TITLE
knownFolders ios downloads return undefined

### ISSUE THREAD

**Issue by jitendraP-ashutec** [NativeScript/NativeScript#9522]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI: 8.0.2
 - Cross-platform modules:
 - iOS Runtime:8.0.0
 - XCode Version:12.5.1

**Describe the bug**
I want to implement a feature to download an image and store it in the download folder. I am trying to get the download folder path using `knownFolders.ios.downloads()` code but its returns undefined. I have tried other method like library, pictures but it still return undefined. Could you please suggest how I can I do that.

Source code sample:

```

 ImageSource.fromUrl(imageURL).then(res => {
      const folderDest = knownFolders.ios.downloads();
      const pathDest = path.join(folderDest.path, "test.png");
      const saved = res.saveToFile(pathDest, "png");
      if (saved) { 
          console.log("Image saved successfully!");  
      }
    }, error => {
      console.log(error);
    })

**Comment by jitendraP-ashutec** [comment]
@tsonevn  

I have tried https://github.com/NativeScript/NativeScript/issues/6256 example but it's not working in native script 8.

It is giving me below error:

```====== Assertion failed ======
Native stack trace:
1          0x109c5b9ff tns::Assert(bool, v8::Isolate*) + 119
2          0x109c54617 tns::ClassBuilder::GetTypeEncodingType(v8::Isolate*, v8::Local<v8::Value>) + 121
3          0x109c52642 tns::ClassBuilder::ExposeDynamicMethods(v8::Local<v8::Context>, objc_class*, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::Local<v8::Object>) + 1332
4          0x109c520d3 tns::ClassBuilder::ExposeDynamicMembers(v8::Local<v8::Context>, objc_class*, v8::Local<v8::Object>, v8::Local<v8::Object>) + 219
5          0x109c516ee tns::ClassBuilder::ExtendCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 468
6          0x109d890a1 v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 609
7          0x109d8858a v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) + 570
8          0x109d87bad v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) + 253
9          0x10a65e7b9 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit + 57
JavaScript stack trace:
at saveToAlbum (file: app/app/shared/image-view/image-slideshow.ts:126:40)
at <anonymous> (file: app/app/shared/image-view/image-slideshow.ts:79:20)

**Comment by rigor789** [comment]
Seems like ios no longer creates these folders by default - the paths are returned in the native apis, but core checks if the returned path exists before returning a `Folder` instance.

https://github.com/NativeScript/NativeScript/blob/980eb213f9871e89a62f883bbd9fb540f6370295/packages/core/file-system/index.ts#L632-L646

https://github.com/NativeScript/NativeScript/blob/980eb213f9871e89a62f883bbd9fb540f6370295/packages/core/file-system/file-system-access.ios.ts#L113-L135

`path` is resolved to the correct path to Downloads, however line 116 returns false if the `Downloads` folder does not exist - which is the case by default (not sure what version of ios changed this).

We need to either not check if the folder exists - or refactor `knownFolders` to return the path regardless of it's existence (perhaps an additional boolean indicating if the path exists), or we should create the directory when it doesn't exist and return the path?

@NathanWalker @farfromrefug curious what you think.

**Comment by rigor789** [comment]
Yep the path returned by `NSSearchPathDirectory.DownloadsDirectory` points to the apps own directory - which does not have a `Downloads` folder in there:

![image](https://user-images.githubusercontent.com/879060/153630508-297d4f15-aec1-4d9d-a4af-7db1baa80732.png)

I believe we _can_ create folders here - though would need to test to make sure...
As far as accessing the system `Downloads` folder - I don't think that's possible, or at least not without extra permissions?

**Comment by ray007** [comment]
Isn't the system `Downloads` folder the one all apps should be able to access?
So you can download a file and open (import) it from your own app.
Not sure when this stopped working, but that's the folder I was hoping to access...

**Comment by farfromrefug** [comment]
@ray007 agree this is what i expect too. Will search a bit

**Comment by farfromrefug** [comment]
@rigor789 @ray007 i ran some tests. First i confirm what you say, the download folder is per app and not created by default.
I think we should change [`getKnownPath`](https://github.com/Akylas/NativeScript/blob/master/packages/core/file-system/file-system-access.ios.ts#L376) to return `Folder.fromPath(url.path).path` to fix the creation issue.
Now i tried to see if iOS was creating a `merged` Download folder for all but no :s  i tested it and it is not the case.
It seems there is only `download` folder per app domain.
I actually dont see any reference to `Download` folder from an ios simulator. @ray007 do you know if the user can access a `Download` folder from his phone?

**Comment by ray007** [comment]
@farfromrefug When opening the `Files` app, I can go to `Locations` > `On My iPhone` > `Downloads`, which is where I can also see downloaded files in.
I'd assume (hope) it should be possible to also access this folder from my own app.
But I do not know which path it has and if I would need extra permissions for it.
I do know that some time ago (probably before iOS 15?), this was the downloads-folder I got from `knownFolders`.

**Comment by ray007** [comment]
@farfromrefug I tested on an actual iPhone and downloaded a file with safari to make sure there is one. Quite possible that there isn't one before you actually download anything.

**Comment by ray007** [comment]
So, no more accessing the Downloads folder on iOS?
Does anyone have a workaround for this problem?
